aws/defaults/handlers.go

```diff
@@ -252,3 +269,34 @@ var ValidateEndpointHandler = aws.NamedHandler{Name: "core.ValidateEndpointHandl
 		r.Error = aws.ErrMissingEndpoint
 	}
 }}
+
+// AttemptClockSkewHandler records the estimated clock skew between the client
+// and service response clocks. This estimation will be no more granular than
+// one second. It will not be populated until after at least the first
+// attempt's response is received.
+var AttemptClockSkewHandler = aws.NamedHandler{
+	Name: "core.AttemptClockSkewHandler",
+	Fn: func(r *aws.Request) {
+		if r.ResponseAt.IsZero() || r.HTTPResponse == nil || r.HTTPResponse.StatusCode == 0 {
+			return
+		}
+
+		respDateHeader := r.HTTPResponse.Header.Get("Date")
+		if len(respDateHeader) == 0 {
+			return
+		}
+
+		respDate, err := http.ParseTime(respDateHeader)
+		if err != nil {
+			if r.Config.Logger != nil {
+				r.Config.Logger.Log(fmt.Sprintf("ERROR: unable to determine clock skew for %s/%s API response, invalid Date header value, %v",
+					r.Metadata.ServiceName, r.Operation.Name, respDateHeader))
+			}
+			return
+		}
+
+		r.AttemptClockSkews = append(r.AttemptClockSkews,
+			respDate.Add(-1*time.Second).Sub(r.ResponseAt.Add(-1*time.Second)),
+		)
```

Comment:
What's the purpose of subtracting the one second from each value?

Reply:
I have an outstanding question about that as well from the design doc this is implemented from. I cannot see a reason for it, but wanted to confirm before removing it.

Reply:
Going to remove this for now. Can add it back in if needed.  Best guess is that the design doc used this as some kind of error correction if the Date header wasn't available.